File: ActionUIAndroid/library/src/main/java/com/abracode/actionui/Views/EmptyView.kt
package com.abracode.actionui.Views

import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import com.abracode.actionui.Common.ActionUIElement
import com.abracode.actionui.Common.ActionUIViewConstruction

/**
 * Renders nothing. Mirror of the Apple `EmptyView` element
 * (`ActionUI/Views/EmptyView.swift`), which wraps `SwiftUI.EmptyView`, and the
 * fallback the renderer uses where SwiftUI would substitute an `EmptyView` (e.g.
 * a [Link] / [ShareLink] with no valid URL).
 *
 * An `EmptyView` produces no layout node, so the base [modifier] has nothing to
 * attach to and is intentionally unused - matching the Apple element, where a
 * modifier on an `EmptyView` likewise produces nothing visible.
 */
object EmptyView : ActionUIViewConstruction {
    @Composable
    override fun BuildView(element: ActionUIElement, modifier: Modifier) {
        // Intentionally empty: renders nothing.
    }
}

File: ActionUIWeb/src/Views/EmptyView.js
// EmptyView.js — EmptyView element.
// Web analog of ActionUI/Views/EmptyView.swift (SwiftUI.EmptyView) and
// ActionUIAndroid Views/EmptyView.kt (a no-op).
//
// Renders nothing. The node is an empty, layout-collapsed box (display:none via
// .aui-empty) so it occupies no space and draws nothing, matching SwiftUI's
// EmptyView and the Android no-op. It is still registered (and carries its
// data-aui-id when given one) so a cross-platform document resolves the type and
// a host can address it, but it has no value and no children.
//
// Note: like Apple, baseline View modifiers (padding/background/frame/...) apply
// to nothing here - the element is deliberately invisible. There is no divergence
// worth a Web_Porting_Notes entry; this is the trivial member of the group-A
// batch (EmptyView / Group / Link / ShareLink / ContentUnavailableView).

import { register } from "../Common/ActionUIRegistry.js";

register("EmptyView", {
    valueType: "none",

    // EmptyView.swift validates nothing (no properties); pass through.
    validateProperties: (properties) => properties,

    buildView: () => {
        const node = document.createElement("div");
        node.className = "aui-empty";
        return node;
    },
});
